File: lista_de_exercicios/questao3/questao3.go
package main

type ContaBancaria struct {
    titular string
    saldo   float64
}

func NovaConta(titular string, saldo float64) *ContaBancaria {
    return &ContaBancaria{titular: titular, saldo: saldo}
}

func (c *ContaBancaria) Depositar(valor float64) {
    if valor > 0 {
        c.saldo += valor
    }
}

func (c *ContaBancaria) Sacar(valor float64) {
    if valor > 0 && valor <= c.saldo {
        c.saldo -= valor
    }
}

func (c *ContaBancaria) GetSaldo() float64 {
    return c.saldo
}

File: lista_de_exercicios/questao3/questao3.py
class ContaBancaria:
    def __init__(self, titular, saldo=0):
        self.titular = titular
        self.__saldo = saldo

    def depositar(self, valor):
        if valor > 0:
            self.__saldo += valor

    def sacar(self, valor):
        if 0 < valor <= self.__saldo:
            self.__saldo -= valor

    def get_saldo(self):
        return self.__saldo
